drivers/flash/soc_flash_mcux.c

```diff
@@ -87,6 +159,12 @@ static int flash_mcux_read(struct device *dev, off_t offset,
 	 */
 	addr = offset + priv->pflash_block_base;
 
+#ifdef CONFIG_CHECK_BEFORE_READING
+	if (is_area_readable(addr, len)) {
+		return -EIO;
```

Comment:
Above, you return various errors from `is_area_readable`, which this then just always collapses into `-EIO`.

Reply:
you are right, forgot to update the read function

Reply:
fixed in new push